examples/05-fleet-mine/index.ts

```diff
@@ -0,0 +1,260 @@
+        const resourceTokenFromPubkey = resourceTokenFrom.address;
+        const resourceTokenTo = await createAssociatedTokenAccountIdempotent(mintPubkey, cargoHoldPubkey);
+        const resourceTokenToPubkey = resourceTokenTo.address;
+
+        const [starbasePubkey] = Starbase.findAddress(sageProgram as any, gamePubkey, planet.data.sector as any);
+
+        let ix = Fleet.asteroidMiningHandler(
+            sageProgram as any,
+            cargoProgram as any,
+            fleetPubkey,
+            starbasePubkey,
+            mineItemPubkey,
+            resourcePubkey,
+            planetPubkey,
+            cargoHoldPubkey,
+            ammoBankPubkey,
+            foodCargoTypePubkey,
+            ammoCargoTypePubkey,
+            resourceCargoType,
+            cargoStatsDefinitionPubkey,
+            gameStatePubkey,
+            gamePubkey,
+            foodTokenFromPubkey,
+            ammoTokenFromPubkey,
+            resourceTokenFromPubkey,
+            resourceTokenToPubkey,
+            foodMintPubkey,
+            ammoMintPubkey,
+        );
+
+        let tx = await buildAndSignTransaction(ix, keypairToAsyncSigner(myWallet), { connection: connection });
+        let resp = await sendTransaction(tx, connection);
+        console.log(resp);
+
+        // ---------------------
+        // Stop Mining Asteroid
+        // ---------------------
+        const [fuelCargoTypePubkey] = CargoType.findAddress(cargoProgram as any, cargoStatsDefinitionPubkey, mintPubkey, 0);
+        const fuelTokenFrom = await createAssociatedTokenAccountIdempotent(fuelMintPubkey, fuelTankPubkey);
+        const fuelTokenFromPubkey = fuelTokenFrom.address;
+
+        const pointsCouncilRankXp = game.data.points.councilRankXpCategory as any;
+        const councilRankXpCategoryPubkey = pointsCouncilRankXp.category as PublicKey;
+        const councilRankXpModifierPubkey = pointsCouncilRankXp.modifier as PublicKey;
+        const [councilRankXpUserAccountPubkey] = UserPoints.findAddress(pointsProgram as any, councilRankXpCategoryPubkey, playerPofilePubkey);
+
+        const pointsMiningXp = game.data.points.miningXpCategory as any;
+        const miningXpCategoryPubkey = pointsMiningXp.category as PublicKey;
+        const miningXpModifierPubkey = pointsMiningXp.modifier as PublicKey;
+        const [miningXpUserAccountPubkey] = UserPoints.findAddress(pointsProgram as any, miningXpCategoryPubkey, playerPofilePubkey);
+
+        const pointsPilotXp = game.data.points.pilotXpCategory as any;
+        const pilotXpCategoryPubkey = pointsPilotXp.category as PublicKey;
+        const pilotXpModifierPubkey = pointsPilotXp.modifier as PublicKey;
+        const [pilotXpUserAccountPubkey] = UserPoints.findAddress(pointsProgram as any, pilotXpCategoryPubkey, playerPofilePubkey);
+
+        // https://solscan.io/tx/xMWgBMpiFHeUPw6hkw2k5be3LUuNWcH6UNBkyRbqpGUAe156NpNnScXw1oMmX9n191AzhP8AQsmyGdLuGHgCPJu
+        ix = Fleet.stopMiningAsteroid(
```

Comment:
This is a bit tricky and heavily undocumented. For technical reasons, `asteroidMiningHandler` and `stopMiningAsteroid` are two separate instructions, which have to be called in two subsequent transactions (as you already figured out). 

I think this is worth a comment explaining what's happening here and mentioning that between those two calls, there must not be a delay of more than 300 seconds.